pkg/s3/auth_middleware_test.go

```diff
@@ -14,6 +14,11 @@ import (
 	"github.com/minio/minio-go/v7/pkg/signer"
 )
 
+const (
+	secretKey = "AKIAJIFCAFBEBI"
```

Comment:
I would expect these values to be random from the function?

Reply:
probably changing the package name to be `s3_test` would fix the cycling dependency?

Reply:
Please see the PR Notes:
>Tests in `auth_middleware_test.go` use hardcoded secret key and access key values because if trying to generate them in that file I am getting the `import cycle not allowed in testgo list` error (probably because the tests are in the `watch` package and import `gomega` and `ginkgo` too).

If you know how to fix this warning or think that I should move some tests for key generation to a separate package please let me know.

Reply:
>probably changing the package name to be s3_test would fix the cycling dependency?

Yes, it should work, but we usually try to keep the tests in the same package. But if it's required let's do it.

Are random values good for unit tests in general? Maybe it's better to use a list of pre-generated key values for consistency? Ir is it some sort of an integration test?

Reply:
And if we change the package name can I export the `verifySignature` function use in the tests too? Because it's private now.

Reply:
IMHO it's a much more idiomatic way to have tests in a dedicated *_test package so that you're forced to only test the public API.